src/doc/git-workflow.md

```diff
@@ -1,86 +1,113 @@
+Tips for making your contribution smoothly:
 
-### Develop and contribute an improvement
+- Explain why your change is a good idea using the text area of the Pull Request.
+- Don't rebase your branch after the pull request is open; make corrections by pushing new commits.
+- If your change is a work in progress then prefix the Pull Request name with `[wip]`.
+- If your change is a rough draft for early feedback then prefix the Pull Request name with `[sketch]`.
 
-1. [Create your own fork](https://help.github.com/articles/fork-a-repo/) of Snabb Switch on Github.
-2. Develop and debug your contribution on a new [topic branch](https://git-scm.com/book/en/v2/Git-Branching-Branching-Workflows#Topic-Branches) based on the latest `master`.
-3. Make a final cleanup of your code before review. (Last chance to rebase.)
-4. Submit a Github [Pull Request](https://help.github.com/articles/using-pull-requests/#initiating-the-pull-request)
-   to the `master` branch.
-5. Respond to feedback and correct problems by pushing additional commits.
+### Being a maintainer
 
-There are two milestones in the process of accepting your change:
+So you have decided to help with the upstream maintenance of Snabb
+Switch. Fantastic! You can do this by creating and maintaining a
+"subsystem branch" where you take responsibility for reviewing and
+merging some of the Pull Requests submitted to Snabb Switch.
 
-1. Your change is merged onto a branch that feeds `master`, for
-   example `next`, `fixes`, `documentation-fixes`, or `nfv`. From this
-   point the owner of that branch will push your work upstream
-   together with other related changes. They might ask you for help
-   but otherwise your work is done.
-2. Your change is merged onto `master`. This could happen in a series
-   of merge steps, for example `nfv->next->master`. Once this happens
-   your code has been officially released as part of Snabb Switch.
+#### Registering a subsystem branch
 
-### Develop and maintain a new program
+The first step is to create and register your subsystem branch:
 
-Snabb Switch includes programs like `snabbnfv`, `packetblaster`, and
-`snsh`. Here is how you can create a new program and take charge of
-its development.
+1. Pick your technical area of interest. What kind of changes will you be responsible for reviewing and merging? Try to pick an area that is easy to identify, for example "the packetblaster program", "the Intel I350 device driver", or "the Git Workflow chapter of the manual".
+2. Create a branch with a suitable name on your Github fork, for example `packetblaster`, `i350`, or `git-workflow`. This is where you will merge relevant changes.
+3. Describe this branch in the file `src/doc/branches.md` and open a Github Pull Request. This will kick off two discussions: how to clearly identify the changes that you are responsible for, and to which "next-hop" upstream branch you should send the changes that you have accepted by merging.
```

Comment:
We haven't figured that out completely yet. I think in most cases the PRs should be against master, then again some PRs only make sense targeting a specific branch (e.g. PRs related to the IPsec topic branch for instance). I have to update SnabbBot to merge PRs with their target branch instead of a fixed branch (master in our case) for the latter case to work well.